embedded-graphics/src/primitives/polyline.rs

```diff
@@ -0,0 +1,358 @@
+                    stop: false,
+                    vertices: rest,
+                    translate: self.translate,
+                    segment_iter:ThickLineIterator::new(
+                        &Line::new(*start + self.translate, *end + self.translate),
+                        1,
+                    ),
+                })
+            })
+            .unwrap_or_else(||
+                // Polyline is less than 2 vertices long. Return a dummy iterator that will short
+                // circuit due to `stop: true`
+                Self::IntoIter {
+                    stop: true,
+                    vertices: &[],
+                    translate: Point::zero(),
+                    segment_iter: ThickLineIterator::new(&Line::new(Point::zero(), Point::zero()), 1),
+                })
+    }
+}
+
+impl<'a, C> IntoIterator for &'a Styled<Polyline<'a>, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+    type IntoIter = StyledPolylineIterator<'a, C>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StyledPolylineIterator {
+            style: self.style,
+
+            line_iter: self.primitive.into_iter(),
+        }
+    }
+}
+
+/// Pixel iterator for each pixel in the line
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct StyledPolylineIterator<'a, C>
+where
+    C: PixelColor,
+{
+    style: PrimitiveStyle<C>,
+
+    line_iter: PolylineIterator<'a>,
+}
+
+impl<'a, C: PixelColor> Iterator for StyledPolylineIterator<'a, C> {
+    type Item = Pixel<C>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        // Break if stroke width is zero
+        if self.style.stroke_width == 0 {
+            return None;
+        }
+
+        // Return none if stroke color is none
+        let stroke_color = self.style.stroke_color?;
+
+        self.line_iter
+            .next()
+            .map(|point| Pixel(point, stroke_color))
+    }
+}
+
+impl<'a, C: 'a> Drawable<C> for &Styled<Polyline<'a>, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    fn draw<D: DrawTarget<C>>(self, display: &mut D) -> Result<(), D::Error> {
+        display.draw_iter(self.into_iter())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{geometry::Size, pixelcolor::Rgb565, prelude::*};
+
+    // A "heartbeat" shaped polyline
+    const HEARTBEAT: [Point; 10] = [
+        Point::new(10, 64),
+        Point::new(50, 64),
+        Point::new(60, 44),
+        Point::new(70, 64),
+        Point::new(80, 64),
+        Point::new(90, 74),
+        Point::new(100, 10),
+        Point::new(110, 84),
+        Point::new(120, 64),
+        Point::new(300, 64),
+    ];
+
+    #[test]
+    fn positive_dimensions() {
+        let polyline = Polyline::new(&HEARTBEAT);
+
+        let bb = polyline.bounding_box();
+
+        assert_eq!(bb.top_left, Point::new(10, 10));
+        assert_eq!(bb.top_left + bb.size, Point::new(301, 85));
+        assert_eq!(bb.size, Size::new(291, 75));
```

Comment:
Is this correct? I'm not sure if it's an off-by-one error, or something related with #182.

Reply:
Yes, the size is correct. But I would remove the `bb.top_left + bb.size` test, because that's not the correct way to calculate the bottom right corner of the bounding box.

Reply:
Cool, thanks for confirming. I see `bottom_right` referenced in a few places, but it's not a method or property on `Rectangle`. What's the correct way?

Reply:
The correct way is to use the soon to be added `bottom_right` getter.

But you could also use `assert_eq(bb, Rectangle::with_corners(Point::new(10, 10), Point::new(300, 84))`.